File: api-go/models/transaction.go
package models

type Transaction struct {
	Type             string `json:"type" binding:"required"`
	AccountID        int    `json:"account_id" binding:"required"`
	MerchantID       *int   `json:"merchant_id"`        // Optional: can be null
	ReferenceEventID *int   `json:"reference_event_id"` // Optional: can be null
	AmountCents      int    `json:"amount_cents" binding:"required"`
	Currency         string `json:"currency" binding:"required,len=3"` // Validates exactly 3 chars
}

File: tests/load/transaction.js
import http from 'k6/http';
import { sleep, check } from 'k6';

export const options = {
    stages: [
        { duration: '50s', target: 100 }
    ]
};


export default function () {

    const res = http.post('http://localhost:8080/transaction', JSON.stringify({
        "type": "purchase",
        "account_id": 101,
        "merchant_id": 5005,
        "amount_cents": 4500,
        "currency": "USD"
        }));

    check(res, {
        'is status 201': (r) => r.status == 201,
    });
}